test/postsubmit-tests-with-pipeline-deployment.sh

```diff
@@ -0,0 +1,149 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -x
+
+usage()
+{
+    echo "usage: deploy.sh
+    [--platform             the deployment platform. Valid values are: [gcp, minikube]. Default is gcp.]
+    [--workflow_file        the file name of the argo workflow to run]
+    [--test_result_bucket   the gcs bucket that argo workflow store the result to. Default is ml-pipeline-test
+    [--test_result_folder   the gcs folder that argo workflow store the result to. Always a relative directory to gs://<gs_bucket>/[PULL_SHA]]
+    [--timeout              timeout of the tests in seconds. Default is 1800 seconds. ]
+    [-h help]"
+}
+
+PLATFORM=gcp
+PROJECT=ml-pipeline-test
+TEST_RESULT_BUCKET=ml-pipeline-test
+GCR_IMAGE_BASE_DIR=gcr.io/ml-pipeline-staging/
+TARGET_IMAGE_BASE_DIR=gcr.io/ml-pipeline-test/${PULL_BASE_SHA}
+TIMEOUT_SECONDS=1800
+NAMESPACE=kubeflow
+
+while [ "$1" != "" ]; do
+    case $1 in
+             --platform )             shift
+                                      PLATFORM=$1
+                                      ;;
+             --workflow_file )        shift
+                                      WORKFLOW_FILE=$1
+                                      ;;
+             --test_result_bucket )   shift
+                                      TEST_RESULT_BUCKET=$1
+                                      ;;
+             --test_result_folder )   shift
+                                      TEST_RESULT_FOLDER=$1
+                                      ;;
+             --timeout )              shift
+                                      TIMEOUT_SECONDS=$1
+                                      ;;
+             -h | --help )            usage
+                                      exit
+                                      ;;
+             * )                      usage
+                                      exit 1
+    esac
+    shift
+done
+
+#Variables
+TEST_RESULTS_GCS_DIR=gs://${TEST_RESULT_BUCKET}/${PULL_BASE_SHA}/${TEST_RESULT_FOLDER}
+DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" > /dev/null && pwd)"
+
+echo "postsubmit test starts"
+
+# activating the service account
+gcloud auth activate-service-account --key-file="${GOOGLE_APPLICATION_CREDENTIALS}"
+gcloud config set compute/zone us-central1-a
+gcloud config set core/project ${PROJECT}
+
+#Uploading the source code to GCS:
+local_code_archive_file=$(mktemp)
+date_string=$(TZ=PST8PDT date +%Y-%m-%d_%H-%M-%S_%Z)
+code_archive_prefix="gs://${TEST_RESULT_BUCKET}/${PULL_BASE_SHA}/source_code"
+remote_code_archive_uri="${code_archive_prefix}_${PULL_BASE_SHA}_${date_string}.tar.gz"
+tar -czf "$local_code_archive_file" .
+gsutil cp "$local_code_archive_file" "$remote_code_archive_uri"
+
+TEST_CLUSTER_PREFIX=${WORKFLOW_FILE%.*}
+TEST_CLUSTER=$(echo $TEST_CLUSTER_PREFIX | cut -d _ -f 1)-${PULL_BASE_SHA:0:7}-${RANDOM}
+
+## Wait for the cloudbuild job to be started
```

Comment:
could you extract the code that waits for cloud builder job to a util function and have a environment variable to control whether the code would be executed or not. something like

if [[$USE_CLOUD_BUILDER_IMAGE]]; then
   waitCloudBuilder()
fi
this way the presubmit_test script is reused. 

We can also rename the script to something else, like tests_with_pipeline_deployment.sh